src/adam_bashforth_methods.jl

```diff
@@ -0,0 +1,647 @@
+        #--####################################################################
+
+        ## Calculation of g_{j}(n)coefficients for j=0,...,k+1;
+        g_coefs!(g,c,h[n],tspan,n,min(n,steporder+2))
+
+        ## Calculation of  Φstar_{j}(n), and Φ_{j}(n) coef. for j=0,...,k;
+        Φ_and_Φstar_coefs!(b, Φ, Φstar,dy[n],tspan,n,1:min(max(n-1,1),steporder+1))
+        #Note that  1<= min(max(n-1,1),steporder+1) <= n-1 and steporder+1
+
+        ## Calculate next value, y[n+1], with order = k = steporder + 1. See
+        #equation (5.5) in Hairer et al, Vol 1, where we note that here
+        # y[n+1] = y_{k}(t_{n+1})
+        y[n+1] = y[n]
+        for j= 1:steporder
+          y[n+1] += h[n]*g[j]* Φstar[2,j]
+        end
+
+        # Compute current estimate for derivative
+        dy[n+1] = F(tspan[n+1],y[n+1])
+
+        #--#####################################################################
+        # Correction-Evaluaion (CE) Loop for implicit solver
+        #     See Harier p. 360 for a general description of this method
+        #--#####################################################################
+
+        # Set CE_loops to be how many correction-evaulation cycles
+        # desired. As least one loop is necessary, for the sake of the
+        # adaptive step selection
+        CE_loops = 1
+        loop_counter = 0
+        pred = y[n+1]
+        while loop_counter<CE_loops && n>=steporder+1
+            loop_counter += 1
+            #(C)orrection of y[n+1]
+            #add new coefficients  based on this prediction
+            ##computation of Φ, Φstar using recurrence relations
+            Φ_coefs!(Φ, Φstar,dy[n+1],1:steporder+1)
+
+            ##calculation of g_{steporder + 1}(n) coefficients using recurrence relations
+            ##has already been done above
+
+            ## We correct the computation of y[n+1] using y[n+1]
+            ## as part of our prediction Harier equation 5.7
+            pred = pred + h[n]*g[steporder+1]*Φ[3,steporder+1]
+
+            #(E)valuate estimate for next derivative again
+            dy[n+1] = F(tspan[n+1],pred)
+        end
+
+        y[n+1] = pred # this is y_{k+1}(n+1), implicit approx. of order k+1
+
+        #--#####################################################################
+        # Prepare for next step
+        #--#####################################################################
+
+        # Shift data from this step, from the "current step" position to the "previous step"
+        # position for Φstar and Φ
+        Φstar[1,1:min(max(n-1,1),steporder+1)] = Φstar[2,1:min(max(n-1,1),steporder+1)]
+        Φ[1,1:min(max(n-1,1),steporder+1)] = Φ[2,1:min(max(n-1,1),steporder+1)]
+
+        # Increment our step
+        n= n+1
+    end #while loop over steps
+    return vcat(tspan), y
+end
+
+################################################################################
+# Implicit Fixed Step Adam-Bashforth solver
```

Comment:
Not clear what the difference is to `ode_ab_fsfo`.  I think, at most include one non-adaptive solver.